tests_and_analysis/test/euphonic_test/test_force_constants_calculate_qpt_phonons_castep.py

```diff
@@ -0,0 +1,492 @@
+            np.array([  # Qpoints
+                [-1.00, 9.35, 3.35],
+                [-1.00, 9.00, 3.00]
+            ])
+        ),
+        (
+            "graphite", "graphite.castep_bin",
+            np.array([  # Qpoints
+                [0.00, 0.00, 0.00],
+                [0.001949, 0.001949, 0.00],
+                [0.50, 0.00, 0.00],
+                [0.25, 0.00, 0.00],
+                [0.00, 0.00, 0.50]
+            ])
+        )
+    ]
+
+    kwargs = [
+        {"use_c": False, "fall_back_on_python": True, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 1},
+        {"use_c": True, "fall_back_on_python": False, "n_threads": 2},
+        {
+            "use_c": False, "fall_back_on_python": True,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 1, "asr": "realspace"
+        },
+        {
+            "use_c": True, "fall_back_on_python": False,
+            "n_threads": 2, "asr": "realspace"
+        }
+    ]
+
+    @pytest.fixture(
+        params=list(itertools.product(materials, kwargs))
+    )
+    def calculate_qpoint_phonon_modes(self, request):
+        material, kwargs = request.param
+        material_name, castep_bin_file, qpts = material
+        filename = os.path.join(self.path, material_name, castep_bin_file)
+        return ForceConstants.from_castep(filename), qpts, material_name, kwargs
+
+    def test_calculate_qpoint_phonon_modes(self, calculate_qpoint_phonon_modes):
+        fc, qpts, material_name, kwargs = calculate_qpoint_phonon_modes
+        print(material_name)
+        qpoint_phonon_modes = fc.calculate_qpoint_phonon_modes(
+            qpts, **kwargs
+        )
+        if "asr" in kwargs:
+            test_expected_freqs = self.expected_freqs[material_name]["asr"]
+        else:
+            test_expected_freqs = self.expected_freqs[material_name]["no_asr"]
+        npt.assert_allclose(
+            qpoint_phonon_modes.frequencies.to('hartree').magnitude,
+            test_expected_freqs.to('hartree').magnitude,
+            atol=1e-10
+        )
+
+    @pytest.mark.parametrize(
+        ("material_name", "castep_bin_file", "fc_mat_asr_file"),
+        [
+            ('graphite', 'graphite.castep_bin', 'graphite_fc_mat_asr.npy'),
+            ('LZO', 'La2Zr2O7.castep_bin', 'lzo_fc_mat_asr.npy')
+        ]
+    )
+    def test_enforce_realspace_acoustic_sum_rule(
+            self, material_name, castep_bin_file, fc_mat_asr_file):
+        castep_bin_filepath = os.path.join(
+            self.path, material_name, castep_bin_file
+        )
+        fc = ForceConstants.from_castep(castep_bin_filepath)
+        expected_fc_mat = np.load(
+            os.path.join(self.path, material_name, fc_mat_asr_file)
+        )
+        fc_mat = fc._enforce_realspace_asr()
+        npt.assert_allclose(fc_mat, expected_fc_mat, atol=1e-18)
```

Comment:
`_enforce_realspace_asr` is a private function so we don't need to test it (sorry I know it was already tested in the legacy tests but we should probably not reproduce it)

Reply:
Ok, removed